ksqldb-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4

```diff
@@ -387,6 +395,9 @@ TUMBLING: 'TUMBLING';
 HOPPING: 'HOPPING';
 SIZE: 'SIZE';
 ADVANCE: 'ADVANCE';
+RETENTION: 'RETENTION';
+GRACE: 'GRACE';
+PERIOD: 'PERIOD';
```

Comment:
I think we should add these to `nonReservedWords` - since these can only occur in the window clause it should be safe for users to use them without backquoting them as columns

Reply:
Rest of the WINDOW keywords are reserved. So wondering if we can tease this apart as a separate issue.. wdyt.. Happy either way

Reply:
`YEAR | MONTH | DAY | HOUR | MINUTE | SECOND` are non reserved, which is interesting... but I'm happy to punt this to another time.

Reply:
yes HOPPING, TUMBLING etc are .. Raised #4773 (nice coincidence with the issue number)